src/TiciTacaToeyGameEngine.ts

```diff
@@ -392,6 +461,47 @@ class TiciTacaToeyGameEngine implements GameEngine {
       case MessageTypes.START_GAME:
       case MessageTypes.JOIN_GAME:
       case MessageTypes.SPECTATE_GAME:
+      case MessageTypes.UPDATE_TIME: {
```

Comment:
Wow this seems like a crazy amount of code being duplicated from the make_move section below. Can we try to extract out common bits and refactor please?